app/org/sagebionetworks/bridge/models/schedules/ScheduleStrategy.java

```diff
@@ -15,8 +18,32 @@
 })
 public interface ScheduleStrategy {
     
+    /**
+     * Get a specific schedule for the given user. This can vary by user, based on the strategy 
+     * (the implementation for assigning schedules), however, it needs to be idempotent (each 
+     * call for a user must return the same schedule), and it must be possible to enumerate all 
+     * possible schedules that can be returned by this strategy.
+     * @param studyIdentifier
+     * @param plan
+     * @param user
+     * @return
+     */
     public Schedule getScheduleForUser(StudyIdentifier studyIdentifier, SchedulePlan plan, User user);
     
+    /**
+     * Validate that the strategy implementation instance is valid.
+     * @param errors
+     */
     public void validate(Errors errors);
+    
+    /**
+     * Get all possible schedules that this schedule strategy might schedule. This can be used to 
+     * examine or manipulate all the schedules regardless of the strategy implementation. It must 
+     * be possible to enumerate all possible schedules that can be returned by this strategy.
+     * 
+     * @return an immutable list of all possible schedules that can be generated by this strategy.
+     */
+    @JsonIgnore
+    public List<Schedule> getAllPossibleSchedules();
```

Comment:
Pretty sure annotations aren't inherited by subclasses. Can we explicitly test serialization and verify whether allPossibleSchedules is showing up in the JSON?